src/lyra/abstract_domains/numerical/sign_domain.py

```diff
@@ -0,0 +1,541 @@
+"""
+Sign Abstract Domain
+====================
+
+Abstract domain to be used for **sign analysis**. The set of possible values of a program variable
+is represented as an element of the sign lattice that keeps track whether the variable may be
+negative, positive, or zero.
+
+:Author: Jérôme Dohrau
+"""
+from collections import defaultdict
+from copy import deepcopy
+
+from lyra.abstract_domains.lattice import ArithmeticMixin
+from lyra.abstract_domains.state import State
+from lyra.abstract_domains.store import Store
+from lyra.core.expressions import *
+from lyra.core.types import FloatLyraType, ListLyraType
+
+from lyra.core.utils import copy_docstring
+
+
+class SignLattice(ArithmeticMixin):
+    """
+    Sign lattice.
+    """
+
+    def __init__(self, negative=True, positive=True, zero=True):
+        super().__init__()
+        self._negative = negative
+        self._positive = positive
+        self._zero = zero
+
+    @property
```

Comment:
Should this be a property or not? I wasn't sure about this. Maybe you have some strong opinions on this.

Reply:
No, I would not make these a property.